simulator/func_sim/t/unit_test.cpp

```diff
@@ -112,21 +58,3 @@ TEST_CASE( "FuncSim: Register size")
     CHECK( FuncSim<MIPS32>().sizeof_register() == bytewidth<uint32>);
     CHECK( FuncSim<MIPS64>().sizeof_register() == bytewidth<uint64>);
 }
-
-TEST_CASE( "Run_SMC_trace: Func_Sim")
```

Comment:
Please put tests cases back.

Reply:
Should I comment them?

Reply:
No, you have to make tests working without the removed classes.

Reply:
But what am I supposed to do if I don't have trace initialization any more? Should I implement `init_trace` method into `FuncSim` and `PerfSim` classes' interfaces?

Reply:
For example:

```c++
CHECK_NOTHROW( FuncSimAndMemory<MIPS32>().run_trace_no_limit( smc_code));
```

becomes:
```c++
FuncSimAndMemory<MIPS32> sim;
auto mem = generate_hierarchied_memory();
ElfLoader elf( smc_code);
elf.load_to( mem.get());
sim.set_memory( mem);
sim.set_pc( elf.get_startPC());

CHECK_NOTHROW( sim.run_no_limit() );
````